pkg/image/resolve.go

```diff
@@ -240,38 +278,41 @@ func fetchImagesFromCollection(obj map[string]interface{}) (images []string) {
 	return images
 }
 
-func getRequirementImages(osType OSType) []string {
+func setRequirementImages(osType OSType, imagesSet map[string]map[string]bool) {
+	coreLabel := "core"
 	switch osType {
 	case Linux:
-		return []string{
-			"busybox",
-			settings.ShellImage.Get(),
-		}
+		addSourceToImage(imagesSet, settings.ShellImage.Get(), coreLabel)
+		addSourceToImage(imagesSet, "busybox", coreLabel)
 	}
-	return []string{}
 }
 
-func normalizeImages(rawImages []string) []string {
-	var images []string
-
-	// mirror
-	for i := range rawImages {
-		rawImages[i] = img.Mirror(rawImages[i])
-	}
-
+func generateImageAndSourceLists(imagesSet map[string]map[string]bool) ([]string, []string) {
+	var images, imagesAndSources []string
 	// unique
-	var imagesSet = map[string]bool{}
-	for _, image := range rawImages {
-		if _, exist := imagesSet[image]; !exist {
-			imagesSet[image] = true
-			images = append(images, image)
-		}
+	for image := range imagesSet {
+		images = append(images, image)
 	}
 
 	// sort
 	sort.Slice(images, func(i, j int) bool {
 		return images[i] < images[j]
 	})
 
-	return images
+	for _, image := range images {
+		imagesAndSources = append(imagesAndSources, fmt.Sprintf("%s %s", image, getSourcesList(imagesSet[image])))
+	}
+
+	return images, imagesAndSources
+}
+
+func getSourcesList(sources map[string]bool) string {
+	commaSeparatedSources := ""
+	for source, val := range sources {
+		if !val {
+			continue
+		}
+		commaSeparatedSources += fmt.Sprintf(",%s", source)
+	}
+	return strings.TrimPrefix(commaSeparatedSources, ",")
```

Comment:
```suggestion
	var commaSeparatedSources []string
	for source, val := range sources {
		if !val {
			continue
		}
		commaSeparatedSources = append(commaSeparatedSources, source)
	}
	return strings.Join(commaSeparatedSources, ",")

```

Reply:
sure, here is an excerpt of the new rancher-images-sources.txt:
```
rancher/externalip-webhook:v0.1.6 rancher-external-ip-webhook:0.1.601-rc01
rancher/flannel-cni:v0.3.0-rancher6 rke
rancher/fleet-agent:v0.3.4-rc7 fleet:0.3.400-rc07,fleet-agent:0.3.400-rc07
rancher/fleet:v0.3.4-rc7 fleet:0.3.400-rc07
rancher/fluentd:v0.1.24 rancher-logging:0.2.4
rancher/gitjob:v0.1.13 fleet:0.3.400-rc07
rancher/gke-operator:v1.0.1-rc1 rancher-gke-operator:1.0.100
rancher/grafana-grafana:6.7.4 rancher-istio:1.5.901
rancher/grafana-grafana:7.1.5 rancher-monitoring:0.2.1
rancher/hyperkube:v1.16.15-rancher1 rke
```

Reply:
ok thank you. I think if we can sort based on source it would look better? so that images are grouped by their source